core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -653,7 +653,7 @@ private object RTS {
           if (io eq null) done(value.asInstanceOf[ExitResult[E, A]])
           else evaluate(io)
 
-        case ExitResult.Failed(t, _) => evaluate(IO.fail[E](t))
+        case ExitResult.Failed(t, ts) => evaluate(IO.fail0[E](t, ts))
```

Comment:
It's wonderful to see these errors no longer thrown away. You caught all the places we were throwing away errors? I could just recall this one off the top of my head.

Reply:
I haven't changed the signature of `redeem` (multiple combinators rely on it), I think this is the only place left.